### DISCUSSION TITLE
How are previews for faces determined?

### DISCUSSION THREAD

**Discussion by v-marinkov** [bpatrik/pigallery2#646] [Q&A]
What determines which headshot is used for the thumbnail preview of a particular person ("face")? Can we manually control it?

**Comment by bpatrik** [comment]
The latest photo of a person will be selected: https://github.com/bpatrik/pigallery2/blob/68d09cec469524359ceb91f001a8c4fccf6cf3c8/src/backend/model/database/PersonManager.ts#L34-L44

**Comment by hajo62** [comment]
The previews are a part of the picture and not the faces which has been tagged?
<img width="230" alt="Screenshot" src="https://github.com/bpatrik/pigallery2/assets/15800980/0870a746-fc23-401d-a9f2-33c08d83b508">
I see some face previews (on the faces overview page), where the shown preview does not show the tagged person, although the image itself shows that the tag is set at the correct area.

**Comment by bpatrik** [comment]
I'm not sure if I understand the question.

Face regions are from the exif/xmp data that is the first few bytes of a
jpg file.

**Comment by hajo62** [comment]
In my case the face region is specified with digicam and stored in the xmp meta data in the file. 
If I show the image the face area is displayed correct. 
But when I go to faces, the screenshot above is shown, which is another area in the same image.
![image](https://github.com/bpatrik/pigallery2/assets/15800980/d59e2ce4-5736-4e17-acb5-d29dd574436d)

I have other _cases_ where a wrong face is shown in the faces overview page.

**Comment by bpatrik** [comment]
Did you updated that photo?

Can you try deleting the faces folder in the temp folder, maybe it fixes it?

-- Sorry for being brief, sent from my phone.

**Comment by hajo62** [comment]
No updates, except metadata with digikam.

Will try deleting the faces temp-folder.

**Comment by bpatrik** [comment]
I suspect that you had a photo with the wrong face region coordinates. The
app used that to generate the person thumbnail. Then you fixed the face
region, so on the lightbox/photo previous it works.

(Face region is the xmp tag name in the photo metadata)

-- Sorry for being brief, sent from my phone.

**Comment by hajo62** [comment]
Hhm, might be. 
But that would not explain why - in another case - a wrong face (i.e. _my_ face on _your_ tag thumbnail) is shown in the faces overview page.

Tomorrow I will try deleting the faces temp-folder and look what happens.

**Comment by hajo62** [comment]
I deleted the f-folder and restarted the container.
Unfortunately no change.

Regarding "wrong face region coordinates and fixed later": You never know, but I assume not. The region would have been far away from the correct face region.

How would I best push pigallery to rescan and update the face tags for an image or a folder?

**Reply by bpatrik** [reply]
If you want to be extra sure delete the DB and recreate it. (And then also delete the faces temp folder).

You can also hit the reset db button on the settings page that should also work.

**Comment by beber56** [comment]
Hello, 
same pb for me; face region is specified with digikam ;
PiGallery2 edge-debian-bullseye, docker version;

in face overview, wrong region is showed;
On the photo, it's correct;
I suspect it's when the photo is flipped in exif data;
in exif :

[IFD0]          Orientation                     : Rotate 180
[...]
[XMP-mwg-rs]    RegionAppliedToDimensionsW      : 4032
[XMP-mwg-rs]    RegionAppliedToDimensionsH      : 2688
[XMP-mwg-rs]    RegionAppliedToDimensionsUnit   : pixel
[XMP-mwg-rs]    RegionName                      : Domi, Isa
[XMP-mwg-rs]    RegionType                      : Face, Face
[XMP-mwg-rs]    RegionAreaX                     : 0.596726, 0.338294
[XMP-mwg-rs]    RegionAreaY                     : 0.71596, 0.703497
[XMP-mwg-rs]    RegionAreaW                     : 0.139881, 0.12996
[XMP-mwg-rs]    RegionAreaH                     : 0.236235, 0.311756
[XMP-mwg-rs]    RegionAreaUnit                  : normalized, normalized

I can send you the photo if you want ...

![20230908_131519](https://github.com/bpatrik/pigallery2/assets/105120783/4cabc47c-3454-44e6-9302-a89e6c76f438)
![20230908_131552](https://github.com/bpatrik/pigallery2/assets/105120783/a49a11d6-4990-45fe-821c-6bde935b2ff7)

**Comment by bpatrik** [comment]
Hi
Yes pls. Send a photo that has the issue.

Patrik

-- Sorry for being brief, sent from my phone.

**Comment by beber56** [comment]
hello,

attached ...

Le ven. 8 sept. 2023 à 13:35, Patrik J. Braun ***@***.***> a
écrit :

> Hi
> Yes pls. Send a photo that has the issue.
>
> Patrik
>
> -- Sorry for being brief, sent from my phone.
>

**Comment by beber56** [comment]
![1](https://github.com/bpatrik/pigallery2/assets/105120783/07f8b930-e4e3-44bd-a6a2-a843b8a7be58)

**Comment by bpatrik** [comment]
[7ec6725](https://github.com/bpatrik/pigallery2/commit/7ec6725c59cfbb057b160dd667836c08da8daf84) should fix this.

**Comment by beber56** [comment]
Yes !
I confirm, after removing directory pigallery2/f/

Thanks !
